design-proposals/edge-infra-cve-tracking.md

```diff
@@ -0,0 +1,184 @@
+- `fixed_cves`: a list of CVEs that was fixed in the given OS Profile Version.
+  This is filled from information provided by the OS vendor.
+
+**Instance**:
+- `cves`: a list of CVEs that affects the given Instance at runtime. The
+  information on this field comes from the Edge Node itself. For Immutable OSes
+  this should contain the same values as the `affected_cves` from the
+  `current_os` OS Profile.
+
+Since we support both Immutable OSes, in particular the Edge Microvisor Toolkit
+(EMT), and mutable OSes, in particular Ubuntu, we need to design support for CVE
+reporting for both OSes. Unfortunately, there is no standard way to publish CVEs
+report, so this feature needs to be designed and implemented on a per-OS basis.
+
+### OS Profile
+
+For the OS profile, for both `affected_cves` and `fixed_cves` we expect support
+from the OS Vendor or Provider to provide these information.
+
+For EMT, we expect further information to be published in the Release Service
+together with the OS Image itself, and the Manifest:
+- A structured file with information of existing CVEs of a given EMT version.
+  This file needs to be kept updated by the team maintaining EMT to properly
+  publish information of new CVEs affecting a certain EMT version. CVEs notice
+  should be published until support for that OSProfile is ended.
+- Release Notes for a given EMT version. This file contains the list of fixed
+  CVEs on the given version. The Release Notes should be published in a
+  structured way, so that it can be parsed and used by the orchestrator. For
+example Google Container-Optimized OS published the list of Fixed CVEs for each
+versions on every release, for example
+https://cloud.google.com/container-optimized-os/docs/release-notes/m121. OS
+Resource Manager (OSRM) can populate information about the fixed CVEs when
+creating the OSProfile. For existing CVEs instead it must periodically query the
+Release Service to get the published file, and fill those information into the
+OSProfile.
+
+For Ubuntu, instead, there are multiple ways to get these information:
+- [Ubuntu CVE Tracker](https://ubuntu.com/security/cves): this is a web
+  interface to navigate CVEs, you can filter by Ubuntu versions, and check the
+  status of the CVE.
+- [CVE git repo](https://git.launchpad.net/ubuntu-cve-tracker/) Canonical
+  maintain a public git repo with structured CVE data
+- [Ubuntu OVAL](https://ubuntu.com/security/oval) Ubuntu provides vulnerability
+  feeds as structured XML file. This contains all CVEs affecting Ubuntu
+  releases, both for packages provided in the Image and on other OS components.
+- [osv.dev](https://osv.dev) Ubuntu participates in OSV.dev, that provides a
+  standardized CVE APIs. However, this is able to provide CVE for packages, not
+  general CVEs affecting a certain Ubuntu release, for example affecting the
+  kernel.
+
+The best choice needs to be agreed with a PoCs, this should cover both
```

Comment:
Since we don't have A/B update for Ubuntu, the only way to fix CVEs is to perform mutable OS update. This will effectively mean that "fixed_cves" in the Ubuntu OS profiles will only be informational and only applied for freshly provisioned ENs. 

I'm ok with that, just making a note that the approach to fix CVEs will be different and in the case of immutable OS we apriori know which CVEs will be fixed, while for mutable OSes it depends on runtime operations made to OS.

Reply:
Yes, that's true. And that should be the behaviour of fixed CVEs for mutable OS. We can report information only about the image itself, not of CVEs that will be fixed later by upgrading the node.

Reply:
Added this comment in the proposal